### DISCUSSION TITLE
Attack Map doesn't update

### DISCUSSION THREAD

**Discussion by gran-pitaya** [telekom-security/tpotce#1393] [Q&A]
I have currently deployed Tpotce in standalone mode. The only issue I am facing is that the Attack Map doesn't update to current or previous attacks. The Kibana dashboard is updating just fine. So, I don't know what could be causing this problem at the moment (all docker containers are up, based on dps.sh) I've only assigned about 10Gb of RAM due to my own resource limitation currently. I may be able to increase that in a bit. I checked the logs for map_redis, map_web and map_data (Bc of another comment from a year ago) all containers seems normal except for map_redis. 

By running "docker logs map_redis" it shows a warning "WARNING Memory overcommit must be enabled! Without it background save or replication may fail under low memory condition. to fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl  vm.overcommit_memory=1' for this to take affect.". The issue is I don't know what this change would do to the machine, as I don't want to break it.

What else should I be checking for?
Anyone who had similar issues, your opinion/guidance will be appreciated. Thanks!

**Comment by Ganesh-403** [comment]
@gran-pitaya 
Since your **Kibana dashboards are updating correctly**, Elasticsearch and the ingestion pipeline are most likely working. That narrows the issue down to the **Attack Map components** (`map_data`, `map_web`, or `map_redis`) rather than the honeypots or ELK stack.

Regarding the Redis warning:

```text
WARNING Memory overcommit must be enabled!
vm.overcommit_memory = 1
```

This is a **standard Redis recommendation**. It does **not** mean Redis has failed—it warns that background save or replication *may* fail under certain memory conditions. Enabling `vm.overcommit_memory=1` is a common recommendation for Redis deployments and is generally safe on Linux because it changes how the kernel handles memory allocation requests rather than immediately consuming more memory. :contentReference[oaicite:0]{index=0}

You can apply it temporarily:

```bash
sudo sysctl vm.overcommit_memory=1
```

Or persist it across reboots:

```bash
echo "vm.overcommit_memory=1" | sudo tee -a /etc/sysctl.conf
sudo sysctl -p
```

That said, **I don't think this warning alone explains why the Attack Map isn't updating**, especially since Redis is still starting normally.

Instead, I'd recommend checking the following:

1. **Is `map_data` receiving events?**

```bash
docker logs -f map_data
```

You should see it periodically processing attack events. If there are no new events, the problem is likely between Elasticsearch and `map_data`.

2. **Can `map_web` communicate with `map_redis`?**

```bash
docker logs map_web
docker logs map_redis
```

Look for connection errors, exceptions, or repeated reconnect attempts.

3. **Verify the Attack Map configuration**

Recent T-Pot versions expose Attack Map settings through the `.env` file, including:

```text
TPOT_ATTACKMAP_TEXT=ENABLED
```

If this has been changed or disabled, it could affect how the Attack Map components behave. :contentReference[oaicite:1]{index=1}

4. **Available memory**

While 10 GB RAM is below the recommended amount for a full standalone T-Pot deployment, the fact that Kibana and Elasticsearch are functioning suggests you're not experiencing a complete resource exhaustion issue. However, if Elasticsearch is under heavy memory pressure, `map_data` could lag behind.

One question that would help narrow this down:

- Is the Attack Map **completely empty**, or does it display historical attacks but stop receiving new ones?
- Also, could you share the last 20–30 lines of:

```bash
docker logs map_data
docker logs map_web
docker logs map_redis
```

Those logs should indicate whether the issue is with event retrieval, Redis communication, or the web frontend.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
